src/qt/walletcontroller.cpp

```diff
@@ -147,23 +146,64 @@ void WalletController::removeAndDeleteWallet(WalletModel* wallet_model)
     delete wallet_model;
 }
 
+WalletControllerActivity::WalletControllerActivity(WalletController* wallet_controller, QWidget* parent_widget)
+    : QObject(wallet_controller)
+    , m_wallet_controller(wallet_controller)
+    , m_parent_widget(parent_widget)
+{
+}
 
-OpenWalletActivity::OpenWalletActivity(WalletController* wallet_controller, const std::string& name)
-    : m_wallet_controller(wallet_controller)
-    , m_name(name)
-{}
+WalletControllerActivity::~WalletControllerActivity()
+{
+    delete m_progress_dialog;
```

Comment:
Could naked `delete` be avoided in the new code?

Reply:
Ownership is given to `m_parent_widget` so that it shows up properly. If the activity finishes first then we delete it - note lines L174-L176 below.

Reply:
See: https://github.com/bitcoin/bitcoin/pull/16261#discussion_r303228603